File: src/macos/AzPin/Models/PinnedResource.swift
import SwiftData

@Model
final class PinnedResource {
    var id: String
    var name: String
    var type: String
    var resourceGroup: String
    var subscriptionId: String
    var location: String
    var displayOrder: Int

    init(id: String, name: String, type: String, resourceGroup: String, subscriptionId: String, location: String, displayOrder: Int) {
        self.id = id
        self.name = name
        self.type = type
        self.resourceGroup = resourceGroup
        self.subscriptionId = subscriptionId
        self.location = location
        self.displayOrder = displayOrder
    }
}

File: src/windows/AzPin.Windows.Core/Models/Entities/PinnedResource.cs
using System.ComponentModel.DataAnnotations;

namespace AzPin.Windows.Models.Entities;

public class PinnedResource
{
    [Key]
    public int LocalId { get; set; }
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public string ResourceGroup { get; set; } = string.Empty;
    public string SubscriptionId { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public int DisplayOrder { get; set; }
    public int? PinnedResourceGroupLocalId { get; set; }
    public PinnedResourceGroup? PinnedResourceGroupEntity { get; set; }
}
